orchagent/vxlanorch.cpp

```diff
@@ -385,27 +453,57 @@ remove_tunnel_termination(sai_object_id_t term_table_id)
     }
 }
 
+//------------------- VxlanTunnel Implementation --------------------------//
+
+VxlanTunnel::VxlanTunnel(string name, IpAddress srcIp, IpAddress dstIp, tunnel_creation_src_t src)
+                :tunnel_name_(name), src_ip_(srcIp), dst_ip_(dstIp), src_creation_(src)
+{
+   VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+
+   if(dstIp.isZero())
+   {
+     //tunnel_orch->addVTEP(std::unique_ptr<VxlanTunnel>this,srcIp);
+     tunnel_orch->addVTEP(this,srcIp);
+     vtep_ptr = NULL;
+   }
+   else if(src_creation_ == TNL_CREATION_SRC_EVPN) 
+   {
+     vtep_ptr = tunnel_orch->getVTEP(srcIp);
+     tunnel_orch->addRemoveStateTableEntry(name,srcIp, dstIp,
+                                           src, true);
+   }
+}
+
+VxlanTunnel::~VxlanTunnel()
+{
+   VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+   tunnel_orch->addRemoveStateTableEntry(tunnel_name_,src_ip_, dst_ip_,
+                                          src_creation_, false);
+}
+
 bool VxlanTunnel::createTunnel(MAP_T encap, MAP_T decap, uint8_t encap_ttl)
 {
     try
     {
         sai_ip_address_t ips, ipd, *ip=nullptr;
+        uint8_t mapper_list = 0;
         swss::copy(ips, src_ip_);
 
-        ids_.tunnel_decap_id = SAI_NULL_OBJECT_ID;
-        ids_.tunnel_encap_id = SAI_NULL_OBJECT_ID;
+        // Only a single mapper type is created
+
+        if(decap == MAP_T::VNI_TO_BRIDGE)
+          TUNNELMAP_SET_BRIDGE(mapper_list);
+        else if(decap == MAP_T::VNI_TO_VLAN_ID)
+          TUNNELMAP_SET_VLAN(mapper_list);
+        else
+          TUNNELMAP_SET_VRF(mapper_list);
+        
+        createMapperHW(mapper_list, (encap == MAP_T::MAP_TO_INVALID) ? USE_DECAP_ONLY: USE_DEDICATED_ENCAP_DECAP);
```

Comment:
can we not modify `create_tunnel_map` and return ids as a list?

Reply:
create_tunnel_map is retained as is to encapsuate into the structure required by the SAI API and for 
consistency with other create_tunnel.. calls.